### PR TITLE
Bug 1594111 - Change data source to BigQuery

### PR DESCRIPTION
**Pull request by Ben-Wu**
Everything seems to be working except esr which doesn't work in prod either so I assume it's a non-issue.

Need to make some updates in the deployed postgres:
```sql
UPDATE 
  measure 
SET 
  enabled=false 
WHERE 
  name='browser_shim_usage_blocked'
  OR name='defective_permissions_sql_removed' 
  OR name='permissions_sql_corrupted'
  OR name='slow_script_notice_count' 
  OR name='slow_script_page_count';

UPDATE platform SET telemetry_name = 'Mac' WHERE name = 'mac';

UPDATE platform SET telemetry_name = 'Windows' WHERE name = 'windows';
```

### CODE REVIEW

File: missioncontrol/etl/measure.py [hunk 1]
```diff
@@ -113,14 +102,46 @@ def update_measures(application_name, platform_name, channel_name,
         'os_name': platform.telemetry_name,
         'channel_name': channel_name,
         'min_timestamp': min_timestamp.strftime("%Y-%m-%d %H:%M:%S"),
-        'submission_date': submission_date.strftime("%Y%m%d")
+        'submission_date': submission_date.strftime("%Y-%m-%d")
     }
-    logger.info('Querying: %s', query_template % params)
+
+    bq_query = f'''
```
Comment (wlach) [inline missioncontrol/etl/measure.py]: Nit: I feel like `query_sql` might be a bit more of a descriptive variable name here.

File: missioncontrol/etl/measure.py [hunk 2]
```diff
@@ -113,14 +102,46 @@ def update_measures(application_name, platform_name, channel_name,
...
+    logger.info('Querying: %s', bq_query)
+
+    client = get_bigquery_client()
+    query_job = client.query(query=bq_query)
 
     # bulk create any new datum objects from the returned results
     build_cache = {}
     datum_objs = []
-    for row in raw_query(query_template, params):
+    for row in query_job:
```
Comment (wlach) [inline missioncontrol/etl/measure.py]: Shouldn't this be `query_job.result()`? https://googleapis.dev/python/bigquery/latest/index.html#perform-a-query

Comment (Ben-Wu) [reply missioncontrol/etl/measure.py]: Iterating over the job is possible too: https://cloud.google.com/bigquery/docs/reference/libraries#client-libraries-usage-python.  I think result is implicitly called here.  Do you have any readability preference for this?

**Comment by wlach** [comment]
Also, there are a few test/flake8 errors which should be addressed before this lands.

File: missioncontrol/etl/measure.py (hunk 2 above)
Comment (wlach) [reply missioncontrol/etl/measure.py]: Ah, ok, I think the way you wrote it is fine in that case. Just Google's documentation threw me off.

**Comment by whd** [comment]
> Looks pretty good to me, minus one thing that looks like an oversight. @whd what do you think? In particular, how should we make the service account credentials available to the container? This PR currently specifies a file here:
> 
> https://github.com/mozilla/missioncontrol/pull/376/files#diff-fc1d52d91894465e6fac0a9792c0d4f4R98

I'm going to experiment a bit with this today but the standard variable is [GOOGLE_APPLICATION_CREDENTIALS](https://cloud.google.com/docs/authentication/getting-started#setting_the_environment_variable) which you should use instead.

I'll update when I've got the docker credential mounting ops logic squared away.

**Comment by whd** [comment]
The ops pieces of this are prepared in https://github.com/mozilla-services/cloudops-deployment/pull/3525/files and https://github.com/mozilla-services/cloudops-infra/pull/1609. Aside from switching to `GOOGLE_APPLICATION_CREDENTIALS` (after which the SDK will pick up credentials automatically), this looks fine to me.

**Comment by Ben-Wu** [comment]
@whd I can't see those repos.  Should I be able to?
For the credentials nothing needs to be done here except assume `GOOGLE_APPLICATION_CREDENTIALS` is defined right?  Should we put a fake one in the env file?

**Comment by whd** [comment]
> @whd I can't see those repos. Should I be able to?

You should, but it looks like you're missing from the "everyone" group which has read on the repos. You don't need to worry about those PRs though.

> For the credentials nothing needs to be done here except assume `GOOGLE_APPLICATION_CREDENTIALS` is defined right?

Correct.

> Should we put a fake one in the env file?

I don't have an opinion on this. It will be overridden in the deployment config.

**Comment by wlach** [comment]
@Ben-Wu unit tests are failing, I think because we're somehow trying to access GCP directly. Other than that, lgtm

**Comment by Ben-Wu** [comment]
Trying to figure out why it's failing; GCP access should be patched in the test.  It works locally in docker so it's confusing.

**Pull request merged**